applications/ParticleMechanicsApplication/custom_conditions/particle_based_conditions/mpm_particle_base_condition.cpp

```diff
@@ -374,12 +374,81 @@ Matrix& MPMParticleBaseCondition::CalculateCurrentDisp(Matrix & rCurrentDisp, co
     KRATOS_CATCH( "" )
 }
 
+void MPMParticleBaseCondition::CalculateOnIntegrationPoints(
+    const Variable<double>& rVariable,
+    std::vector<double>& rValues,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    if (rValues.size() != 1)
+        rValues.resize(1);
+
+    if (rVariable == MPC_AREA) {
+        rValues[0] = m_area;
+    }
+    else {
+        KRATOS_ERROR << "Variable " << rVariable << " is called in CalculateOnIntegrationPoints, but is not implemented." << std::endl;
+    }
+}
+
+void MPMParticleBaseCondition::CalculateOnIntegrationPoints(const Variable<array_1d<double, 3 > >& rVariable,
+    std::vector<array_1d<double, 3 > >& rValues,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    if (rValues.size() != 1)
+        rValues.resize(1);
+
+    if (rVariable == MP_COORD || rVariable == MPC_COORD) {
```

Comment:
Do we need this distinction between MP_COORD and MPC_COORD as for conditions we exclusively use MPC_COORD and elements MP_COORD ?

Reply:
We were actually thinking of neglecting the distinction completely. It is just introducing bugs and there is no need to do it. But open for discussion?!

Reply:
So, initially, we intend to distinguish all the variables into two sets: MP_VAR and MPC_VAR for material point and boundary points. This is useful if you want to distinguish apply BC for both particle and boundary particles at the same time. So for displacement, velocity, or traction forces, I think it will be better to separate. However, for coordinates, we can generalize them if you like.

Reply:
You can distinguish already with condition or element.

However, it is discussion, but as our elements are very complex we should consider neglecting using both (for maintenance reasons - and no real disadvantage). Having one Base class will ease many procedures.